azure-quantum/azure/quantum/plugins/cirq/targets/ionq.py

```diff
@@ -0,0 +1,170 @@
+    ):
+        self._client = _IonQClient(workspace)
+        super().__init__(
+            workspace=workspace,
+            name=name,
+            input_data_format=input_data_format,
+            output_data_format=output_data_format,
+            provider_id=provider_id,
+            content_type=content_type,
+            encoding=encoding,
+            **kwargs
+        )
+
+    @staticmethod
+    def _translate_cirq_circuit(circuit) -> Dict[str, Any]:
+        """Translate Cirq circuit to IonQ JSON. If dependencies \
+    are not installed, throw error with installation instructions."""
+        try:
+            from cirq_ionq import Serializer
+
+        except ImportError:
+            raise ImportError(
+                "Missing optional 'cirq_ionq' dependency. \
+    To install run: pip install azure-quantum[cirq]")
+
+        else:
+            return Serializer().serialize(circuit)
+
+    def _to_cirq_job(self, azure_job: "AzureJob") -> "CirqIonqJob":
+        """Convert Azure job to Cirq job"""
+        job_dict = self._client._create_job_dict(azure_job)
+        return CirqIonqJob(client=self._client, job_dict=job_dict)
+
+    def submit(
+        self,
+        program: "cirq.Circuit",
+        name: str = "cirq-job",
+        repetitions: int = 500,
+        **kwargs
+    ) -> "CirqIonqJob":
+        """Submit a Cirq quantum circuit
+
+        :param program: Cirq circuit
+        :type program: cirq.Circuit
+        :param name: Job name
+        :type name: str
+        :param repetitions: Number of shots, defaults to 
+            provider default value
+        :type repetitions: int
+        :return: Azure Quantum job
+        :rtype: Job
+        """
+        serialized_program = self._translate_cirq_circuit(program)
+        metadata = serialized_program.metadata or {}
+        metadata["qubits"] = serialized_program.body["qubits"]
+        # Override metadata with value from kwargs
+        metadata.update(kwargs.get("metadata", {}))
+        azure_job = super().submit(
+            circuit=serialized_program.body,
+            name=name,
+            num_shots=repetitions,
+            metadata=metadata,
+            **kwargs
+        )
+        # Get Job shim to process results using cirq.Result
+        return self._to_cirq_job(azure_job=azure_job)
+    
+    @staticmethod
+    def _to_cirq_result(
+        result: Union[QPUResult, SimulatorResult],
+        param_resolver: cirq.ParamResolverOrSimilarType = cirq.ParamResolver({}),
+        seed: cirq.RANDOM_STATE_OR_SEED_LIKE = None,
+    ) -> "cirq.Result":
+        if isinstance(result, QPUResult):
+            return result.to_cirq_result(params=cirq.ParamResolver(param_resolver))
+        elif isinstance(result, SimulatorResult):
+            return result.to_cirq_result(params=cirq.ParamResolver(param_resolver), seed=seed)
+
+        raise ValueError("Result {result} not supported. \
+            Please pass either a cirq_ionq.results.QPUResult \
```

Comment:
```suggestion
            Expecting either a cirq_ionq.results.QPUResult \
```
The user is not calling this method directly, so asking them to pass certain values is missleading.